tests/expressions_case/tests.py

```diff
@@ -1165,6 +1165,16 @@ def test_m2m_reuse(self):
             lambda x: x[1:]
         )
 
+    def test_empty_q_object(self):
+        """A Q object with an empty condition should be rejected as the
+        conditional expression in a Case()."""
+        with self.assertRaisesMessage(ValueError, 'Empty Q() object'):
+            CaseTestModel.objects.annotate(
+                test_name=Case(When(Q(), then=Value(True)),
+                               default=Value(False),
+                               output_field=models.BooleanField())
+            ).all()
```

Comment:
technically your test only needs to do: `When(Q(), then=Value(True))` :) The queryset isn't even being constructed as the exception is happening before.

Reply:
True. I believe this test was necessary with my original implementation, but I should have updated the test when I moved the check to the constructor.